service/access/server.go

```diff
@@ -429,7 +432,8 @@ func (s *Server) ExecuteScriptAtBlockID(ctx context.Context, in *access.ExecuteS
 	blockID := flow.HashToID(in.BlockId)
 	height, err := s.index.HeightForBlock(blockID)
 	if err != nil {
-		return nil, fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+		err = fmt.Errorf("could not get height for block ID %x: %w", blockID, err)
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
The user provided block is not found
```suggestion
		return nil, status.Error(codes.NotFound, err.Error())
```